docs/sources/flow/reference/components/loki.source.gcplog.md

```diff
@@ -0,0 +1,157 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/loki.source.gcplog
+title: loki.source.gcplog
+---
+
+# loki.source.gcplog
+
+`loki.source.gcplog` retrieves logs from cloud resources such as GCS buckets,
+load balancers, or Kubernetes clusters running on GCP by making use of [Pub/Sub
+subscriptions](https://cloud.google.com/pubsub/docs/subscriber).
+
+The component uses either the 'push' or 'pull' strategy to retrieve log
+entries and forward them to the list of receivers in `forward_to`.
+
+Multiple `loki.source.gcplog` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.source.gcplog "LABEL" {
+  pull {
+    project_id   = "PROJECT_ID"
+    subscription = "SUB_ID"
+  }
+
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.gcplog` supports the following arguments:
+
+Name            | Type                 | Description          | Default | Required
+--------------- | -------------------- | -------------------- | ------- | --------
```

Comment:
Should consider adding a row for push and pull to this table since they are top level block arguments?

Reply:
I am considering a layout like this on the heroku stuff but was wondering what you all thought

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/39704712/213797497-9cb8b395-8d2f-4837-9617-e8b679b5ede3.png">

Reply:
The current convention for listing only attributes in the Arguments section, and have a separate section for inner blocks was to accomodate nested block structures, as well as blocks who can be reused in different parts of the configuration and keep it relatively cleaner.

Maybe worth a conversation with @karengermond and the rest of the squad into how we structure component docs, but I think we shouldn't break consistency with for now.

Reply:
got it, I will update mine to match this style